components/remote_settings/src/client.rs

```diff
@@ -541,6 +541,8 @@ pub struct RemoteSettingsRecord {
     pub attachment: Option<Attachment>,
     #[serde(flatten)]
     pub fields: RsJsonObject,
+    #[serde(default)]
+    pub filter_expression: Option<String>,
```

Comment:
Are there really performance involved if `filter_expression` remains a field reflecting what's on the server?

Reply:
`filter_expression` is already part of `fields`. 

Having it as a separate column is a sort of optimization to avoid doing:

```rust
if records.fields.contains_key("filter_expression")
```

Reply:
How is this resolved? 
What advantage is there in having a dedicated storage column for the `filter_expression`?

Reply:
I resolved it! Makes total sense to me now and you are right, we need to get the `filter_expression` from the `fields`. Or at least, it makes more sense that way. I updated the PR.